providers/aix/host_aix.go

```diff
@@ -0,0 +1,206 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package aix
+
+/*
+#cgo LDFLAGS: -L/usr/lib -lperfstat
+
+#include <libperfstat.h>
+#include <procinfo.h>
+#include <sys/proc.h>
+
+*/
+import "C"
+
+import (
+	"os"
+	"time"
+
+	"github.com/joeshaw/multierror"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/go-sysinfo/internal/registry"
+	"github.com/elastic/go-sysinfo/providers/shared"
+	"github.com/elastic/go-sysinfo/types"
+)
+
+//go:generate sh -c "go tool cgo -godefs defs_aix.go | sed 's/*byte/uint64/g' > ztypes_aix_ppc64.go"
+// As cgo will return some psinfo's fields with *byte, binary.Read will refuse this type.
+
+func init() {
+	registry.Register(aixSystem{})
+}
+
+type aixSystem struct{}
```

Comment:
On other platforms, we use `"github.com/prometheus/procfs"` to have this struct carry a procfs root. We can then do this:

```
content, err := ioutil.ReadFile(h.procFS.path("meminfo"))
```

So we can support other file roots for testing. We'll want to do this here.

Reply:
AIX procfs isn't as developed as the Linux one. There is only a limited amount of information, no meminfo, no cpuinfo, etc. 
That's why we either have to call AIX cmd like "svmon" (it was the first version) or use CGO (this vresion).

Reply:
we don't need to use `prometheus/procfs`, but we should still have a struct with a path root string of some kind, so we can create a fake rootfs for testing and we're not hard-coding entire filepaths like `/proc/version`

Reply:
Would it really be worth it ? It would be useful only to check `getOsInfo()` (because of `/proc/version`. Most of the other information (about both host and processes) are retrieved using C functions. 
For processes, you'll need a true one to test them. Creating a fake `/proc/pid` would be pointless (let alone the fact that you have to recreate the AIX structure where files are binaries and not text file like Linux).